src/Functions/array/arrayIntersect.cpp

```diff
@@ -510,11 +510,12 @@ ColumnPtr FunctionArrayIntersect::execute(const UnpackedArrays & arrays, Mutable
         map.clear();
 
         bool all_has_nullable = all_nullable;
+        bool current_has_nullable = false;
```

Comment:
It's better to declare this variable separately in each loop.
The value after the first loop is not used anywhere, and this is not clear right now.